app/waf/src/services/do.service.ts

```diff
@@ -512,6 +514,19 @@ export class OnboardingManager {
           return resObj[0]['result']['code'] === 200;
         },
         reason => {
+          // Wait for iControl LX restarting
```

Comment:
What's other possible statusCode?

Reply:
That means there's a big possibility to change this code part again when encountering new status need-to-wait in the future. maybe again and again?

Could we ignore details status, wait until timeout or success? Because till now we don't know what kinds of statuses need us to quit. 

Seems restnoded has the ability to recover(restart self), there may be little possibility of quit because of server error(5XX) or do install failed(4xx).

Maybe it's more suitable to write it like this: 

```
if (some quit reason) {
    return Promise.reject('reason')
}
if(other quit) {
    // quit immediately. 
}

// default -> wait..
```

Reply:
Your first two 'if' action are equivalent.
I will propose
```
if (some reason) {
  // wait
}
// default quit immediately.
```
as we used to do, because the code can not enumerate unexpected errors. Under most circumstance, the program need to be broken by the unexpected errors, unless we specify to tolerate them.